controller/src/main/java/io/pravega/controller/store/stream/Cache.java

```diff
@@ -28,33 +31,31 @@
 
     public Cache(final Loader loader) {
         cache = CacheBuilder.newBuilder()
-                .maximumSize(10000)
+                .maximumSize(MAXIMUM_SIZE)
                 .expireAfterWrite(10, TimeUnit.MINUTES)
                 .build(new CacheLoader<String, CompletableFuture<Data>>() {
                     @ParametersAreNonnullByDefault
                     @Override
                     public CompletableFuture<Data> load(final String key) {
-                        CompletableFuture<Data> result = loader.get(key);
-                        result.exceptionally(ex -> {
-                            invalidateCache(key);
-                            return null;
-                        });
-                        return result;
+                        return loader.get(key);
                     }
                 });
     }
 
     public CompletableFuture<Data> getCachedData(final String key) {
-        return cache.getUnchecked(key);
+        return cache.getUnchecked(key).exceptionally(ex -> {
+            System.err.println("shivesh:: invalidating cache because of a previous exception.. key = " 
```

Comment:
This is some temporary logging added for debugging travis issue. 
The code is in a bit of a flux because of the issues i am encountering. I will clean these up soon (hopefully by EOD).